python/src/aiconfig/model_parser.py

```diff
@@ -110,6 +110,7 @@ async def run_batch(
         for params in parameters_list:
             # Create a deep copy of the aiconfig object to prevent mutations that could affect other iterations
             aiconfig_deep_copy = copy.deepcopy(aiconfig)
+            prompt = aiconfig_deep_copy.get_prompt(prompt.name)
```

Comment:
Why would this ever be different than the original?

Reply:
deepcopy recursively creates a deep copy of each object.

The contents of the prompt are the same, but the **Object** itself is not the same. Inside the sdk we operate on the prompt itself. 

For example: 

```
  public async run(
    prompt: Prompt,
    aiConfig: AIConfigRuntime,
    options?: InferenceOptions,
    params?: JSONObject | undefined
  ): Promise<Output | Output[]> {  
  ```
  
`prompt.outputs = something`,  

Mutates the state of the prompt _Object_.